alacritty/src/event.rs

```diff
@@ -495,6 +497,56 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         }
     }
 
+    /// Update the cursor blinking state.
+    #[inline]
+    fn update_cursor_blinking(&mut self) {
+        // Get config cursor style.
+        let mut cursor_style = self.config.cursor.style;
+        if self.terminal.mode().contains(TermMode::VI) {
+            cursor_style = self.config.cursor.vi_mode_style.unwrap_or(cursor_style);
+        };
+
+        // Check terminal cursor style.
+        let terminal_blinking = self.terminal.cursor_style().blinking;
+        let blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking);
+
+        // Update cursor blinking state.
+        self.scheduler.unschedule(TimerId::BlinkCursor);
+        if blinking && self.terminal.is_focused {
+            self.scheduler.schedule(
+                GlutinEvent::UserEvent(Event::BlinkCursor),
+                Duration::from_millis(self.config.cursor.blink_interval),
+                true,
+                TimerId::BlinkCursor,
+            )
+        } else {
+            *self.cursor_hidden = false;
```

Comment:
If you have blinking `On` and then you do `printf "\e[?12l\a"`, it's still blinking, but it shouldn't. However it seems like vte literally does nothing on Blinking cursor mode, so maybe it's fine? Like it neither disables it nor enables it, and blinking can only be changed via `CSI Ps SP q`, maybe we should follow it as well? Have you checked how it works in xterm/urxvt?

Reply:
Just a bug.